data-access/Models/ArtefactModels.cs

```diff
@@ -7,12 +7,16 @@ public class ArtefactModel
 		public string Name { get; set; }
 		public string Mask { get; set; }
 		public uint MaskEditorId { get; set; }
-		public string TransformMatrix { get; set; }
-		public uint TransformMatrixEditorId { get; set; }
+		public float? Scale { get; set; }
+		public float? Rotate { get; set; }
+		public uint? TranslateX { get; set; }
```

Comment:
Are the translations always non-negative? Is there no way we'll want to move an artefact that is on the top left corner of the scroll *a little bit* up or left?

Reply:
Yeah, I had a long discussion with Ingo about this a few years back. He was not in favor of allowing negative values, by the same token he also wanted the coordinate system to have its origin at top,right, with value increasing as you move from right to left (essentially an RTL coordinate system).  I really feared the inevitable coding and data manipulation errors that would result from that, and opted for the (universally?) standard top,left origin used by both the browser and GIS.  
In any event, I am not opposed to allowing negative values (I would just need to change the type in the database).  Indeed I always feared the issue of moving something a little above or to the left of 0, which in this system would mean moving every single other piece of data down and to the right, ugh.  I think for the user, they will perhaps have rulers in view and I really dislike the idea of having negative values there (but this can be managed browser side).  So, basically, the real issue is computational/representational simplicity; do you prefer to have every scroll begin at point 0,0 and to support negative numbers, or do you prefer to have every scroll begin at point 2147483647,2147483647 (the middle value of a 32 bit unsigned int) and disallow negative numbers?  I could even let you use signed 32 bit integers when communicating with the API and add 2147483647 to the value before inserting into the database.  What is your preference?

Reply:
Let me see if I can choose a larger font. Here we go:
# NO! 
Top left at (2^31, 2^31) means nobody, ever, is going to be able to read these coordinates and tell you which is to the left of which. People will constantly need to compare 10 digit numbers in their head - and that's a huge mess.

In Israel when they teach you about graphs and the X and Y axes, the X axis goes from left to right (1 is to the right of 0).  So (0,0) as top right is going to confuse every single person looking at the coordinates - Israeli or not. (0,0) as top left is by far the simplest.

As for negatives - I guess it depends on how user friendly we want to be. Having someone move every single piece on the scroll just to make a little room seems very tedious and not at all friendly. We can do this automatically (and generate a huge amount of database updates), but is that really worth it, instead of just allowing negatives? I don't have a strong opinion hear, I'm just wondering. I do feel very strongly about (0,0) being the top left.

Reply:
Absolutely agreed on 0,0 at top left.  I also have no strong feeling on unsigned/signed ints, I guess on the programming side signed integers maybe make more sense and would provide more meaningful values.  I don't mind making the adjustment in the database.

Reply:
If Ingo feels strongly about non-negatives, let's keep the uints. If at some point when people use the editor they decide they need negative coordinates, we'll just change it.

Oh. Oh. Oh. That is only relevant for translations of artefacts relative to scrolls. Other translations (master image vs. another image, for instance) may very well include negative numbers (the master image is at (100, 100) and the PAM image is at (150, 50), so the translation is (50, -50).

Reply:
Ok.  Indeed, this is only for location of artefacts on the virtual scroll.  Other operations will certainly need negative values as you say.

Reply:
That's OK for this model, but I'd like to use the same DTO for all transformations - artefact on scroll or otherwise. So at least the DTO should have ints and not uints.

Reply:
Ok, I will switch the DTO to use signed ints.  And yes, if we can re-purpose the DTO, that would be nice.

Reply:
I saw that you actually subtract 2^31 from the number returned from the database when creating this int. I thought putting the middle point at (2^31, 2^31) was a joke and not something that you actually did - even if only in the database. It is a *horrible horrible idea*. You basically do allow negative numbers, you just encode them in a way that is so unnatural, we may end up as a DailyWTF.